pennylane/drawer/tape_mpl.py

```diff
@@ -208,12 +213,24 @@ def _add_classical_wires(drawer, cwire_layers, cwire_wires):
             if len_diff > 0:
                 layer_wires += [cwire + drawer.n_wires] * len_diff
             for l, w in zip(layer_ids, layer_wires, strict=True):
+                if l < len(layers) and _ppm_controlled(layers[l]):
+                    l -= drawer.ppm_offset()
                 xs.extend([l, l, l])
                 ys.extend([cwire + drawer.n_wires, w, cwire + drawer.n_wires])
 
             drawer.classical_wire(xs, ys)
 
 
+def _ppm_controlled(layer):
+    last_op = layer[0]
+    last_wire = max(last_op.wires)
+    for op in layer[1:]:
+        if (wire := max(op.wires)) > last_wire:
+            last_wire = wire
+            last_op = op
+    return isinstance(last_op, PauliMeasure)
```

Comment:
```suggestion
    return any(isinstance(op, PauliMeasure) for op in layer)
```
?
Unless I'm missing something...